service/simpleSecretService_test.go

```diff
@@ -0,0 +1,207 @@
+			encryptVal:     secretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			numberOfCallsW: 1,
+			numberOfCallsE: 2,
+			expectedErr:    false,
+		},
+		{
+			name:           "Unsuccessful encrypt Write Secret",
+			encryptVal:     secretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			encryptErr:     errors.New("encrypt Error"),
+			numberOfCallsW: 0,
+			numberOfCallsE: 1,
+			expectedErr:    true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			me := newMockEncoder()
+			defer me.AssertNumberOfCalls(t, "Encrypt", tc.numberOfCallsE)
+			me.On("Encrypt", tc.value, ciphering.PassToSecretKey(tc.secretKey)).Return(tc.encryptVal, tc.encryptErr)
+			me.On("Encrypt", tc.key, ciphering.PassToSecretKey(tc.secretKey)).Return(tc.encryptVal, tc.encryptErr)
+
+			ms := newMockStorage()
+			defer ms.AssertNumberOfCalls(t, "Write", tc.numberOfCallsW)
+			ms.On("Write", tc.encryptVal, tc.encryptVal).Return(tc.writeErr)
+
+			ss := New(ms, me, me)
+			err := ss.WriteSecret("key", tc.value, tc.secretKey)
+			if tc.expectedErr {
+				require.Error(t, err)
+
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestSimpleSecretService_ReadSecret(t *testing.T) {
+
+	var testCases = []struct {
+		name           string
+		readVal        string
+		readErr        error
+		encryptVal     string
+		encryptErr     error
+		decryptVal     string
+		decryptErr     error
+		secretKey      string
+		key            string
+		value          string
+		expectedResult string
+		expectedError  bool
+		numberOfCallsR int
+		numberOfCallsE int
+		numberOfCallsD int
+	}{
+		{
+			name:           "Successful Read Secret",
+			readVal:        secretVal,
+			encryptVal:     secretVal,
+			decryptVal:     notSecretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			expectedResult: notSecretVal,
+			expectedError:  false,
+			numberOfCallsR: 1,
+			numberOfCallsE: 1,
+			numberOfCallsD: 1,
+		},
+		{
+			name:           "Read Secret Encrypt Error",
+			readVal:        secretVal,
+			encryptVal:     secretVal,
+			encryptErr:     errors.New("encryptError"),
+			decryptVal:     notSecretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			expectedResult: "",
+			expectedError:  true,
+			numberOfCallsR: 0,
+			numberOfCallsE: 1,
+			numberOfCallsD: 0,
+		},
+		{
+			name:           "Read Secret read Error",
+			readVal:        secretVal,
+			readErr:        errors.New("ReadError"),
+			encryptVal:     secretVal,
+			decryptVal:     notSecretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			expectedResult: "",
+			expectedError:  true,
+			numberOfCallsR: 1,
+			numberOfCallsE: 1,
+			numberOfCallsD: 0,
+		},
+		{
+			name:           "Read Secret Decrypt Error",
```

Comment:
Check out an [article about test naming](https://medium.com/getground/naming-tests-in-golang-c58c188bb9a1).
That's just one article, you could find more examples.